src/concurrency/async/state-machine/recursion.md

```diff
@@ -0,0 +1,34 @@
+---
+minutes: 3
+---
+
+# Recursion
+
+An async function's future type _contains_ the futures for all functions it
+calls. This means a recursive async functions are not allowed.
```

Comment:
I would clarify this by invoking the analogy with recursive enums: they need an indirection to avoid being infinite-sized types. [Prior to Rust 1.77](https://blog.rust-lang.org/2024/03/21/Rust-1.77.0.html#support-for-recursion-in-async-fn), recursion in `async fn` was forbidden entirely and code had to forgo the `async fn` transform in favor of explicitly returning a `Box<impl Future<Output=...>>` type, but now only "bare" recursion (without an indirection) is forbidden.